src/cedict-service/cedict.js

```diff
@@ -0,0 +1,389 @@
+    if (!configuration.storage.stores.dictionary.volatileStorage.hasOwnProperty('enabled')) throw new Error('enabled option of a volatileStorage tree of a dictionary is missing from a configuration') // eslint-disable-line no-prototype-builtins
+    if (!configuration.storage.stores.dictionary.volatileStorage.hasOwnProperty('indexed')) throw new Error('indexed option of a volatileStorage tree of a dictionary is missing from a configuration') // eslint-disable-line no-prototype-builtins
+    if (!configuration.storage.stores.dictionary.permanentStorage) throw new Error('A permanentStorage tree of a dictionary is missing from a configuration')
+    if (!configuration.storage.stores.dictionary.permanentStorage.hasOwnProperty('enabled')) throw new Error('enabled option of a permanentStorage tree of a dictionary is missing from a configuration') // eslint-disable-line no-prototype-builtins
+    if (!configuration.storage.stores.dictionary.permanentStorage.hasOwnProperty('indexed')) throw new Error('indexed option of a permanentStorage tree of a dictionary is missing from a configuration') // eslint-disable-line no-prototype-builtins
+    if (!configuration.data) throw new Error('Date tree is missing from a configuration')
+    if (!configuration.data.version) throw new Error('Data version is missing from a configuration')
+    if (!configuration.data.revision) throw new Error('Data revision is missing from a configuration')
+    if (!configuration.data.recordsCount) throw new Error('Data records count is missing from a configuration')
+    if (!configuration.data.URI) throw new Error('Data URI is missing from a configuration')
+    if (!configuration.data.chunks || configuration.data.chunks.length === 0) throw new Error('Data chunks are missing from a configuration')
+  }
+
+  /**
+   * Initializes a data object.
+   *
+   * @returns {Promise<undefined> | Promise<Error>} - A promise
+   */
+  init () {
+    return new Promise((resolve, reject) => {
+      this._storage = new CedictPermanentStorage(this._configuration.storage)
+      // `storage.connect()` will create a database if it does not exist yet.
+      return this._storage.connect()
+        .catch((error) => {
+          console.error('Connection to storage cannot be established')
+          reject(error)
+        })
+        .then(() => this._storage.getIntergrityData())
+        .then((storageData) => {
+          /*
+          Integrity data has been returned successfully which means database structure is OK.
+          Let's check if there is a new version of data available on a server.
+           */
+          if (
+            storageData.recordsInMeta !== 1 ||
+            storageData.recordsInDictionary !== this._configuration.data.recordsCount ||
+            storageData.metadata.version !== this._configuration.data.version ||
+            storageData.metadata.revision !== this._configuration.data.revision
+          ) {
+            throw new Error('Storage is outdated')
```

Comment:
An error thrown within `then` handler will be converted into a rejection that will caught in the next `catch` handler.